MN/Algoritmi/QR/Givens/Givens.m

```diff
@@ -0,0 +1,38 @@
+% [USES] Algoritmi-ad-hoc/SST
+function [Q R x] = Givens (A, b)
+  [m n] = size(A);
+  % Gt will be the final rotation matrix (the product of all the others)
+  Gt = eye(m);
+  % iterate through the elements below the main diagonal
+  for k = 1 : n-1
+    for l = k + 1 : m
+      % initialize the matrix for this step
+      G = eye(m);
+      
+      % calculate cos and sin
+      ro = sqrt(A(k, k)^2 + A(l, k)^2);
+      c = A(k, k) / ro;
+      s = - A(l, k) / ro;
+      
+      % modify the rotation matrix
+      G(k, k) = c;
+      G(k, l) = -s;
+      G(l, k) = s;
+      G(l, l) = c;
+      
+      % eliminate A(l, k)
+      A = G * A;
+      
```

Comment:
newline-ul asta ar trebui eliminat fiindca eliminate A(l, k) si Gt = G * Gt sunt in aceeasi grupare logica.